cmd/secrets/backend_selector.go

```diff
@@ -0,0 +1,57 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// +build secrets
+
+package secrets
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"time"
+)
+
+type backendPrefix string
+
+const (
+	file       backendPrefix = "file"
+	kubeSecret backendPrefix = "kube_secret"
```

Comment:
nit: not a blocker but given that `backend` is ambiguous, I would probably opt for `secretsProvider` instead of `backend` for clarity. The same principle would apply to the related var names/filenames/implementations.

Reply:
At first, I chose `kube` because we have many metric tags that start with `kube`, but I guess you're right, k8s is more common. I've renamed it :+1:

I also think that "provider" is better than "backend". I've renamed it :+1: